comixed-webui/src/app/comic-files/components/comic-file-covers/comic-file-covers.component.ts

```diff
@@ -16,25 +16,68 @@
  * along with this program. If not, see <http://www.gnu.org/licenses>
  */
 
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnInit, ViewChild } from '@angular/core';
 import { ComicFile } from '@app/comic-files/models/comic-file';
 import { LoggerService } from '@angular-ru/logger';
 import { Store } from '@ngrx/store';
+import { MatMenuTrigger } from '@angular/material/menu';
+import {
+  clearComicFileSelections,
+  setComicFilesSelectedState
+} from '@app/comic-files/actions/comic-file-list.actions';
 
 @Component({
   selector: 'cx-comic-file-covers',
   templateUrl: './comic-file-covers.component.html',
   styleUrls: ['./comic-file-covers.component.scss']
 })
 export class ComicFileCoversComponent implements OnInit {
+  @ViewChild(MatMenuTrigger) contextMenu: MatMenuTrigger;
+
   @Input() files: ComicFile[] = [];
-  @Input() selectedFiles: ComicFile[] = [];
+
+  contextMenuX = '';
+  contextMenuY = '';
+  file: ComicFile = null;
+  fileSelected = false;
+  hasSelections = false;
 
   constructor(private logger: LoggerService, private store: Store<any>) {}
 
+  private _selectedFiles = [];
+
+  get selectedFiles(): ComicFile[] {
+    return this._selectedFiles;
+  }
+
+  @Input() set selectedFiles(selectedFiles: ComicFile[]) {
+    this._selectedFiles = selectedFiles;
+    this.hasSelections = selectedFiles.length > 0;
+  }
+
   ngOnInit(): void {}
 
   isFileSelected(file: ComicFile): boolean {
     return this.selectedFiles.includes(file);
   }
+
+  onShowContextMenu(file: ComicFile, x: string, y: string): void {
+    this.logger.trace('Popping up context menu for:', file);
+    this.file = file;
+    this.contextMenuX = x;
+    this.contextMenuY = y;
+    this.contextMenu.openMenu();
+  }
+
+  onSetOneSelected(selected: boolean): void {
+    this.logger.trace('Setting file selected:', selected);
+    this.store.dispatch(
+      setComicFilesSelectedState({ files: [this.file], selected })
+    );
+  }
+
+  onDeselectAll(): void {
+    this.logger.trace('Selecting all comic files');
```

Comment:
Shouldn't this say 'Deselecting all comic files'  ?

Reply:
Correct. Fixed.

Reply:
You're correct. I'll fix this